src/main/java/com/iota/iri/service/validation/impl/TransactionSolidifierImpl.java

```diff
@@ -0,0 +1,276 @@
+     */
+    private static final int MAX_SIZE= 10000;
+
+    private static final IntervalLogger log = new IntervalLogger(TransactionSolidifier.class);
+
+    /**
+     * Executor service for running the {@link #processTransactionsToSolidify()}.
+     */
+    private SilentScheduledExecutorService executorService = new DedicatedScheduledExecutorService(
+            "Transaction Solidifier", log.delegate());
+    /**
+     * Interval that the {@link #processTransactionsToSolidify()}  will scan at.
+     */
+    private static final int RESCAN_INTERVAL = 500;
+    /**
+     * A queue for processing transactions with the {@link #checkSolidity(Hash)} call. Once a transaction has been
+     * marked solid it will be placed into the {@link #transactionsToBroadcast} queue.
+     */
+    private BlockingQueue<Hash> transactionsToSolidify = new ArrayBlockingQueue<>(MAX_SIZE);
+
+    /**
+     * A set of transactions that will be called by the {@link TransactionProcessingPipeline} to be broadcast to
+     * neighboring nodes.
+     */
+    private BlockingQueue<TransactionViewModel> transactionsToBroadcast = new ArrayBlockingQueue<>(MAX_SIZE);
+    /**
+     * A set containing the hash of already solidified transactions
+     */
+    private Queue<Hash> solidified = new CircularFifoQueue<>(MAX_SIZE);
+
+
+
+    /**
+     * Constructor for the solidifier.
+     * @param tangle                    The DB reference
+     * @param snapshotProvider          For fetching entry points for solidity checks
+     * @param transactionRequester      A requester for missing transactions
+     */
+    public TransactionSolidifierImpl(Tangle tangle, SnapshotProvider snapshotProvider, TransactionRequester transactionRequester){
+        this.tangle = tangle;
+        this.snapshotProvider = snapshotProvider;
+        this.transactionRequester = transactionRequester;
+    }
+
+    /**
+     *{@inheritDoc}
+     */
+    @Override
+    public void start(){
+        executorService.silentScheduleWithFixedDelay(this::processTransactionsToSolidify, 0, RESCAN_INTERVAL,
+                TimeUnit.MILLISECONDS);
+
+    }
+
+    /**
+     *{@inheritDoc}
+     */
+    @Override
+    public void shutdown() {
+        executorService.shutdownNow();
+    }
+
+    /**
+     *{@inheritDoc}
+     */
+    @Override
+    public void addToSolidificationQueue(Hash hash){
+        try{
+            if(transactionsToSolidify.size() >= MAX_SIZE - 1){
+                transactionsToSolidify.remove();
+            }
+
+            if(!solidified.contains(hash) && !transactionsToSolidify.contains(hash)) {
+                    transactionsToSolidify.put(hash);
+            }
+        } catch(Exception e){
+            log.error("Error placing transaction into solidification queue",e);
+        }
+    }
+
+    @Override
+    public boolean addMilestoneToSolidificationQueue(Hash hash, int maxToProcess){
+        try{
+            TransactionViewModel tx = TransactionViewModel.fromHash(tangle, hash);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unnecessary use of fully qualified name 'TransactionViewModel.fromHash' due to existing static import 'com.iota.iri.controllers.TransactionViewModel.fromHash'](https://app.codacy.com/app/GalRogozinski/iri/pullRequest?prid=4339340)